Context, proptypes, ref

Safwood opened the pull request:
![Screenshot_1](https://user-images.githubusercontent.com/69139691/100582674-a700ea80-32fa-11eb-8afb-932b86bc8a07.jpg)
![Screenshot_4](https://user-images.githubusercontent.com/69139691/100582676-a8321780-32fa-11eb-9855-c08ad33f1a47.jpg)
![Screenshot_5](https://user-images.githubusercontent.com/69139691/100582678-a8caae00-32fa-11eb-8892-25a4d0bd7608.jpg)
Добавила контекст, proptypes, ref. Изменила логику переключения страниц, отстилизовала страницы и добавила боковую секцию на страницу регистрации и логина. Кнопки "зарегистрироваться" и "войти" работают и без контекста, остальные кнопки переключатся только, если правильные логин и пароль (mail@mail.ru, 111). После входа появляется хедер с кнопкой "выйти", при нажатии контекст меняется на LoggedOut и появляется страница входа.

Safwood committed:
feat(components): login-context, ma,pbox,  proptypes

Add login-logout context, add mapbox via ref, add proptypes, change logic of switching between
pages.

Safwood committed:
file delete

Allarepossible commented on src/components/Login.jsx:
```diff
@@ -1,26 +1,78 @@
+                </form>
+            </div>
+            ) : (
+              <div className="Form-container">
```
лучше вынести в отдельный компонент

Allarepossible commented on src/components/AuthContext.jsx:
```diff
@@ -0,0 +1,41 @@
+export const AuthProvider = ({children}) => {
+  const [isLoggedIn, setIsLoggedIn] = React.useState(false);
+
+  const LogIn = (email, password) => {
```
функции лучше писать с маленькой, только компоненты и классы пишем с большой

Allarepossible commented on src/components/Map.jsx:
```diff
@@ -1,9 +1,78 @@
+                <img src={logo}  className="Header__logo" alt='logo'/>
+              </div>
+              <nav>
+                <ul className="Header__column">
```
лучше вынести header в отдельный компонент, чтобы не дублировать код

Safwood committed:
fix(components): correct mistakes in Header Create new components for Login

Create LoginForm and LoginNotification components, Create Header component and create function of
coloring button onClick.

Safwood committed:
add redux
